lib/tests/dynamicRow.validation.test.js

```diff
@@ -0,0 +1,85 @@
+import { extractSubElementId } from "../validation";
+import { dynamicRowData as dynForm } from "./testData";
+
+const mockDynaRowFormRes = {
+  3: [
+    {
+      0: "Artisti",
+    },
+  ],
+  7: [
+    {
+      0: "test1",
+      1: "test@cds-snc.ca",
+    },
+  ],
+  12: [
+    {
+      0: "",
+      1: "",
+      2: "",
+      4: {
+        file: null,
+        src: null,
+        name: "",
+        size: 0,
+      },
+      5: {
+        file: null,
+        src: null,
+        name: "",
+        size: 0,
+      },
+      6: "",
+      7: {
+        file: null,
+        src: null,
+        name: "",
+        size: 0,
+      },
+      8: {
+        file: null,
+        src: null,
+        name: "",
+        size: 0,
+      },
+      9: "",
+    },
+  ],
+};
+
+describe("Dynamic rows validator", () => {
+  test("Parses form values data", () => {
+    expect(mockDynaRowFormRes.length === 3);
+  });
+});
+
+describe("extractSubElementId: dynamic rows sub's ids validator", () => {
+  test("Should extract element id ", () => {
+    const elem = dynForm.formConfig.elements.find((element) => element.id === 3);
+    expect(
+      extractSubElementId(elem.properties.subElements.find((subElem) => subElem.id === 4)) === "0"
```

Comment:
Why not pass in the data you expect directly rather than doing all these find operations

Reply:
Do not use boolean operators in expect statements. The syntax here should be `expect(...).toEqual("0")`

Reply:
this syntax throw an error like "toEqual is not a function" etc ... when testing  reason why.

Reply:
Here is a list of matchers jest supports https://jestjs.io/docs/expect